pkg/controllers/awsloadbalancercontroller/rbac_role.go

```diff
@@ -0,0 +1,125 @@
+package awsloadbalancercontroller
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
+)
+
+func (r *AWSLoadBalancerControllerReconciler) ensureRole(ctx context.Context, controller *albo.AWSLoadBalancerController) error {
+	reqLogger := log.FromContext(ctx)
+
+	desired := desiredRole(ctx, r.Namespace)
+	reqLogger.Info("ensuring roles", "roles", desired.Name)
+
+	if err := controllerutil.SetControllerReference(controller, desired, r.Scheme); err != nil {
+		return fmt.Errorf("failed to set the controller reference for roles %s : %w", desired.Name, err)
+	}
+
+	exist, current, err := r.currentRole(ctx, desired)
+	if err != nil {
+		return fmt.Errorf("failed to fetch current roles: %w", err)
+	}
+
+	if !exist {
+		if err := r.createRole(ctx, desired); err != nil {
+			return err
+		}
+
+		_, current, err = r.currentRole(ctx, desired)
+		if err != nil {
+			return fmt.Errorf("failed to fetch created roles: %w", err)
+		}
+		reqLogger.Info("created roles", "roles", desired.Name)
+	}
+
+	if err := r.updateRole(ctx, current, desired); err != nil {
+		return fmt.Errorf("failed to update roles: %w", err)
+	}
+
+	return nil
+}
+
+// updateRoles updates the current roles and returns a flag to denote if the update was done.
+func (r *AWSLoadBalancerControllerReconciler) updateRole(ctx context.Context, current *rbacv1.Role, desired *rbacv1.Role) error {
+	changed, err := isRoleChanged(current, desired)
+	if err != nil {
+		return err
+	}
+
+	if !changed {
+		return nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
+	updated.Rules = desired.Rules
+
+	if err := r.Client.Update(ctx, updated); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isRoleChanged(current *rbacv1.Role, desired *rbacv1.Role) (bool, error) {
+	if current.Name != desired.Name {
+		return true, nil
+	}
+
+	if current.Namespace != desired.Namespace {
+		return true, nil
+	}
+
```

Comment:
Comparing name and namespace is not required.

Reply:
Addressed.